apps/financial-aid/backend/src/app/modules/staff/staff.controller.ts

```diff
@@ -26,17 +17,7 @@ export class StaffController {
     type: StaffModel,
     description: 'Gets staff by nationalId',
   })
-  async getStaffByNationalId(
-    @Param('nationalId') nationalId: string,
-  ): Promise<Staff> {
-    const staff = await this.staffService.findByNationalId(nationalId)
-
-    if (!staff) {
-      throw new NotFoundException(
-        `Staff with nationalId ${nationalId} not found`,
-      )
-    }
-
-    return staff
+  async getStaffByNationalId(@Param('nationalId') nationalId: string) {
+    return await this.staffService.findByNationalId(nationalId)
```

Comment:
Why are you throwing away the NotFoundException? 🤔

Reply:
The reason is because now this is matched with a foreign key both in finding all applications and fetching an application by id. See here: https://github.com/island-is/island.is/pull/5059/files#diff-f3a07a64c5f74919ec24c99d9eeab8d431534a2eed62b62d6b952c66d1378fceR64

It would be fine to throw the error for getting all applications but not with finding an application by id because that is used by Osk as well, what we need todo in the near future is to add another guard or add to the application guard to check if veita members have access to this application depending on the municipality id.

Commented here: https://github.com/island-is/island.is/pull/5059/files#diff-aace4f8af6f227d904d3e9b152ba253fe376bc9b338a7fc02e4cce7f15da28cdR13